### DISCUSSION TITLE
Bug report: The tool suggests line by line where it sould suggest the whole snippet of code

### DISCUSSION THREAD

**Discussion by dudifrid** [community/community#7389] [Copilot Conversations]
In many occassions, the tool knows exactly what I want, but it suggests it line by line, rather than suggesting the whole snippet of code at once. I might find it reasonable where it suggests many suggestions for each line, so the whole snippet could have as many suggestions as the product of the number of suggestions of all the lines. However, it also happens where it suggests only one suggestion to each line of code, and it suggests the next line of code right after I accept the previous line of code, so it sould have been a suggestion of the whole snippet of code at once, and not line by line.

**Comment by Urethramancer** [comment]
This needs to get more consistent, and perhaps configurable for categories of completion.

If the suggestion is a full function or function body, default to spitting out the whole thing. If it's a switch statement not all of its responses may be appropriate, so that's where line mode fits better. But we all sure would like it to at least behave the same every time :)
